src/functions/Mock.ps1

```diff
@@ -1018,6 +1026,11 @@ function FindMatchingBehavior {
     if ($PesterPreference.Debug.WriteDebugMessages.Value) {
         Write-PesterDebugMessage -Scope Mock "No parametrized or default behaviors were found filter."
     }
+
+    if (-not $fallbackAllowed) {
+        throw 'Not implemented. None of the parameterized mocks match the call or accept fallback to original command.'
```

Comment:
For some reason this fails existing `Mocking with nested Pester runs.Mocking works in nested run` test found here https://github.com/pester/Pester/blob/0d3dd6681883f8db2c183a392952e5be383ba1e0/tst/functions/Mock.Tests.ps1#L3119-L3147

It fails because the inner run's `outer mock is not available` calls `Get-Date` which has an existing mock hock from the outer run with Debug logging enabled, while the inner run does not. Result: `Write-PesterDebugMessage` throws in the condition guard.

It tries to print "Mock bootstrap function Get-Date called from block Begin" which is the first line in the mock hook, way before the changed logic. What am I missing?